### PR TITLE
[sanitizer_common] Return nullptr from ASan on ERROR_COMMITMENT_LIMIT

### PR DESCRIPTION
**Pull request by glandium**
Followup to #117929

### CODE REVIEW

**Comment by glandium** [comment]
Cc: @davidmrdavid @zmodem

**Comment by davidmrdavid** [comment]
Since the PR description is empty, just calling out that this is related to: https://github.com/llvm/llvm-project/pull/117929#discussion_r1881342609

**Review by vitalybuka — APPROVED** [review]

File: compiler-rt/lib/sanitizer_common/sanitizer_win.cpp [hunk 1]
```diff
@@ -176,12 +176,12 @@ static void *ReturnNullptrOnOOMOrDie(uptr size, const char *mem_type,
...
-  // Note - It's possible that 'ERROR_COMMITMENT_LIMIT' needs to be handled here
-  // as well. It is currently not handled due to the lack of a reproducer that
-  // induces the error code.
+  // 3. ERROR_COMMITMENT_LIMIT:
+  //  VirtualAlloc will return this if e.g. the pagefile is too small to commit
+  //  the requested amount of memory.
   if (last_error == ERROR_NOT_ENOUGH_MEMORY ||
-      last_error == ERROR_INVALID_PARAMETER)
+      last_error == ERROR_INVALID_PARAMETER ||
+      last_error == ERROR_COMMITMENT_LIMIT)
```
Comment (vitalybuka) [inline compiler-rt/lib/sanitizer_common/sanitizer_win.cpp:184]: would be nice to extend a test if it's reasonably easy to reproduce

Comment (glandium) [reply compiler-rt/lib/sanitizer_common/sanitizer_win.cpp:184]: I haven't found a trivial way to trigger it, unfortunately. At least not in the time I had to try.

**Comment by glandium** [comment]
Could you merge this for me?

**Comment by zmodem** [comment]
>  is that a failure above ^ running on Windows ( I can't see to find any evidence of that)? Otherwise, I can't see why the PR would cause it, since the PR is windows specific.

It's unrelated. That bot is not running Windows, and the test was not using any sanitizer.

**Pull request merged**